### PR TITLE
Parse survey serializer to presenter

### LINKED ISSUE #445: Improve Survey Serializer

### PR DESCRIPTION
**Pull request by edimossilva**
fix #445

This is a WIP, just did it by the feeling, not sure yet if this is the way that we should go :(
Maybe some gem could be helpful, maybe no need, suggestions?

- [x] **Model testing.**
- [x] **Request testing.**
- [x] **System testing.**
- [ ] **No tests required.**

### CODE REVIEW

File: backend/app/presenters/survey_presenter.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  def simple_payload
+    {
+      id: @survey.id,
+      name: @survey.name,
+      description: @survey.description,
+      total_questions_quantity: @survey.questions.size,
+      answered_questions_quantity: @survey.last_answers_quantity
+    }
+  end
```
Comment (garciajordy) [inline backend/app/presenters/survey_presenter.rb:26]: WDYT of giving another parameter in the initializer for example 
```
def initialize(survey, user = nil, simple = false)
```
and put the payloads in private methods and call them in the payload method depending on the simple param?

**Review by garciajordy — CHANGES_REQUESTED** [review]

File: backend/app/presenters/survey_presenter.rb (hunk 1 above)
Comment (edimossilva) [reply backend/app/presenters/survey_presenter.rb:26]: This way, we would make behavior relies more on state. 
Option 1
``` 
survey_presenter = SurveyPresenter (survey, user, true)
survey_presenter.payload

survey_presenter = SurveyPresenter (survey, user, false)
survey_presenter.payload

# con: payload method is smart, it decides things
# con: payload method has multiple responsibilities. It decides the payload and creates a payload (we can solve this by extracting `simple_payload.merge({.....})` to another method)
# con: for having 2 payloads it does need 2 instances 
# pro: only one public method
```

Option 2
```
survey_presenter = SurveyPresenter (survey, user)
survey_presenter.payload(simple: true)
survey_presenter.payload(simple: false)
# con: payload method is smart, it decides things
# con: payload method has multiple responsibilities. It decides the payload and creates a payload (we can solve this by extracting `simple_payload.merge({.....})` to another method)
# pro: only one public method
```

Option 3
```
survey_presenter = SurveyPresenter (survey, user)
survey_presenter.payload
survey_presenter.simple_payload
# con: 2 public methods 
# pro: it doesn't need a logic to decide things
```

Please share your thoughts , do you agree with pros and cons?

@amiapmorais and @iurynogueira , share your thoughts as well

note: I think it is good for us to decide these OO principles

Comment (garciajordy) [reply backend/app/presenters/survey_presenter.rb:26]: I agree with most... but not sure if the logic in the payload could be a con... i've been reading about this and they said if it is something small to add it is ok, but if you are making to many changes you should have another presenter...  I just don't think adding this second public method is the right way, im not very experienced with this so see this thought as a personal opinion over a usefull answer

Comment (edimossilva) [reply backend/app/presenters/survey_presenter.rb:26]: I think the concept behind your supporting is "interface oriented programming".  Usually devs that use dynamic languages are not familiar with this concept.

I agree with following that, I will make the changes soon!

**Pull request merged**